### PR TITLE
fix bugs in shift_from_file and drop_from_file

### PR DESCRIPTION
**Pull request by berkuva**

### CODE REVIEW

File: bedshift/bedshift.py [hunk 1]
```diff
@@ -228,9 +226,10 @@ def _shift(self, row, mean, stdev):
             # check if the region is shifted out of chromosome length bounds
             return None, None
 
-        return row, {0: chrom, 1: start + theshift, 2: end + theshift, 3: "S"}
+        return row, {0: chrom, 1: start + theshift, 2: end + theshift, 3: 'S'}
+
 
-    def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter="\t"):
+    def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter='\t'):
```
Comment (aaron-gu) [inline bedshift/bedshift.py:238]: Can we not undo the format changes from 9ea3ec58c9d69fae7d8da6e20e1964501bd6d76b?

File: bedshift/bedshift.py [hunk 2]
```diff
@@ -247,16 +246,24 @@ def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter="\t"):
         num_shift = int(rows * shiftrate)
 
         intersect_regions = self._find_overlap(fp)
-        interlen = len(intersect_regions)
+        rows2shift = intersect_regions.sample(frac=shiftrate)
+        indices2shift = []
+        for _, r in rows2shift.iterrows():
```
Comment (aaron-gu) [inline bedshift/bedshift.py]: Here is a method that is probably more performant: 

image


Basically to find the common rows, you perform a merge. By using `reset_index` the merge will keep the indices from `self.bed` in the output, which you can then pass to `self.shift`

File: bedshift/bedshift.py (hunk 1 above)
Comment (nsheff) [reply bedshift/bedshift.py:238]: you can also just run `black` on the file to re-create them if that helps @hyunjaecho94 . it's deterministic.

File: bedshift/bedshift.py [hunk 3]
```diff
@@ -247,16 +246,24 @@ def shift_from_file(self, fp, shiftrate, shiftmean, shiftstdev, delimiter="\t"):
...
         if num_shift > interlen:
-            _LOGGER.warning(
-                "Number of regions to be shifted ({}) is larger than the provided bedfile size ({}). Shifting {} regions.".format(
-                    num_shift, interlen, interlen
-                )
-            )
-            num_shift = interlen
-        rows2shift = random.sample(list(range(interlen)), num_shift)
-        return self.shift(shiftrate, shiftmean, shiftstdev, rows2shift)
+            _LOGGER.warning("Number of regions to be shifted ({}) is larger than the desired number of regions to be shifted: ({}). Shifting {} regions.".format(num_shift, interlen, interlen))
```
Comment (aaron-gu) [inline bedshift/bedshift.py]: This new warning message doesn't make sense. It should be more like "Desired regions shifted {} is greater than the number of overlaps found {}."

Also, if `indices2shift` is larger than `num_shift`, it should be downsampled so we are not shifting too many rows.

**Pull request closed without merging**